### PR TITLE
Update xcube prune

### PR DESCRIPTION
**Pull request by forman**
The `xcube prune` tool is no longer restricted to data cube datasets and should now be able to deal with datasets that comprise very many chunks. (#469)

FYI @AliceBalfanz @TonioF 

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] New/modified features documented in `docs/source/*`
* [x] Changes documented in `docs/CHANGES.md`
* [x] AppVeyor and Travis CI passes
* [x] Test coverage remains or increases (target 100%)
* [x] Associated issues closed after merge

### CODE REVIEW

File: xcube/cli/prune.py [hunk 1]
```diff
@@ -21,71 +21,114 @@
...
-@click.argument('cube')
+@click.argument('dataset_path', metavar='DATASET')
+@click.option('--verbose', '-v', is_flag=True, multiple=True,
+              help='Verbose mode. Multiple may be given, '
+                   'for example "-vvv".')
 @click.option('--dry-run', is_flag=True,
-              help='Just read and process input, but don\'t produce any outputs.')
-def prune(cube, dry_run):
+              help='Just read and process input, '
+                   'but don\'t produce any outputs.')
```
Comment (pont-us) [inline xcube/cli/prune.py:41]:
```suggestion
                   'but don\'t produce any output.')
```

File: xcube/cli/prune.py [hunk 2]
```diff
@@ -21,71 +21,114 @@
...
+def prune(dataset_path: str,
+          verbose: Tuple[bool, ...],
+          dry_run: bool):
     """
     Delete empty chunks.
-    Deletes all data files associated with empty (NaN-only) chunks in given CUBE,
+    Deletes all data files associated with empty
+    (NaN-only) chunks in given DATASET,
     which must have ZARR format.
```
Comment (pont-us) [inline xcube/cli/prune.py:49]:
```suggestion
    which must have Zarr format.
```

File: xcube/cli/prune.py [hunk 3]
```diff
@@ -21,71 +21,114 @@
...
         try:
             os.remove(block_path)
             return True
         except OSError as e:
-            monitor(f'error: failed to delete block file {block_path}: {e}')
+            monitor(f'Error: failed to delete '
+                    f'block file {block_path}: {e}', 0)
     else:
-        monitor(f'error: could neither find block file {block_path_1} nor {block_path_2}')
+        monitor(f'Error: could neither find '
```
Comment (pont-us) [inline xcube/cli/prune.py:132]:
```suggestion
        monitor(f'Error: could find neither '
```

**Review by pont-us — CHANGES_REQUESTED** [review]
Change requests are only optional minor improvements to wording in output strings.

I added two more commits -- one to merge master into this branch, and one to fix tests that were failing because they were using chunks which have now been pruned in the repo.

**Comment by forman** [comment]
> I added two more commits -- one to merge master into this branch, and one to fix tests that were failing because they were using chunks which have now been pruned in the repo.

Great, thanks!

**Review by pont-us — APPROVED** [review]

**Pull request merged**